scipy/special/stirling2.h

```diff
@@ -49,7 +56,6 @@ double _stirling2_dp(double n, double k){
             for (int j = 1; j < k; j++){
                 curr[j] = (j + 1) * curr[j] + curr[j - 1];
                 if (isinf(curr[j])){
-                    free(curr);
```

Comment:
Might as well fix this while touching this file: here and below, on overflow, call `sf_error` with the appropriate arguments:

```
                    sf_error("stirling2", SF_ERROR_OVERFLOW, NULL);
```

Reply:
It turns out that this condition can't happen, because currently `_stirling2_dp(n, k)` is called only if `n <= 50`, and for values of `n` that small, there is no overflow.  The change was still worthwhile, since how the code is called could change in the future.